### PR TITLE
Convert project to manifest v3.

### PR DESCRIPTION
**Pull request by colossatr0n**
Additional:
* Use async/await where possible for new manifest v3 service workers.
* Add number of URLs read to options pages.
* Add eslint and chrome typings.
* Update README and remove references to colored links.
* Move icons into their own directory.
* Prettify JSON exports.

Resolves issue https://github.com/edoreld/mark-as-read/issues/19

### CODE REVIEW

**Comment by edoreld** [comment]
Hi and thank you for your PR. 

I find that PRs that focus on one thing are easier to review. Ideally, the PR should focus on upgrading the manifest to V3 and nothing else. That being said, I found some breaking functionality when testing this version. Specifically, the checkbox is ticked off no matter what page is visited.

**Comment by edoreld** [comment]
Upon further testing, the problem seems to be solved. However, I still the PR a bit hard to parse as it is, with so many changes across the board.

**Comment by colossatr0n** [comment]
I imagine most of the hard to parse changes are going to be part of `background.js` with all the async/await code. Manifest v3 uses service workers instead of background pages. Service workers are event driven and don't persist like backgound pages do, so the architecture had to be reworked for v3. See https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/.

This link might help in explaining a lot of the changes in `background.js`: https://developer.chrome.com/docs/extensions/mv3/migrating_to_service_workers/#state

In the meantime, I'll try to rework the PR and see if I can get more of a one-to-one comparison and remove anything extra.

**Comment by colossatr0n** [comment]
Opening a new PR. Closing this one.
